packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -335,64 +264,34 @@ const EDiscovery = WebexPlugin.extend({
+    if (this.contentSummaryBySpaceIdCache.has(reportUUID) && this.contentSummaryBySpaceIdCache.get(reportUUID).has(spaceId)) {
+      return {body: this.contentSummaryBySpaceIdCache.get(reportUUID).get(spaceId), statusCode: 200};
     }
 
     this.logger.warn(`Cache miss for space ${spaceId} in getContentSummaryBySpaceId`);
 
     return this.request({
       method: 'GET',
-      service: 'ediscovery',
-      resource: `reports/${reportId}/contents/summary/${spaceId}`,
+      ...requestOptions,
       timeout: options.timeoutMs
     })
       .then(async (res) => {
-        await this._writeSpacesToContentSummaryBySpaceIdCache(reportId, [res.body]);
+        await this._writeSpacesToContentSummaryBySpaceIdCache(reportUUID, [res.body]);
 
         return Promise.resolve(res);
       });
   },
 
-  @waitForValue('@')
-  @oneFlight({keyFactory: (reportUrl, spaceId) => String(reportUrl + spaceId)})
   /**
-   * Retrieves information for a specific conversation relevant to a specified report
-   * @param {String} reportUrl - full destination address (including report id parameter) for the http request being sent
-   * e.g. 'http://ediscovery-intb.wbx2.com/ediscovery/api/v1/reports/3b10e625-2bd5-4efa-b866-58d6c93c505c'
-   * @param {UUID} spaceId - Id of the conversation being requested
-   * @param {Object} options - optional parameters for this method
-   * @param {number} options.timeoutMs - connection timeout in milliseconds, defaults to 30s
-   * @returns {Promise<ResponseEntity<ContentSummary>>} Http response containing the specified content summary
+   * Wipe the cache used by eDiscovery to store the space summaries. Good practice to clear it down
+   * when finished with a report to save RAM.
+   * @returns {undefined}
    */
-  getRemoteContentSummaryBySpaceId(reportUrl, spaceId, options) {
-    if (!reportUrl || !spaceId) {
-      throw Error('Undefined parameter(s)');
-    }
-
-    const reportId = this._getReportIdFromUrl(reportUrl);
-
-    if (!reportId) {
-      throw Error('Report url does not contain a report id');
-    }
-
-    // use spread operator to set default options
-    options = {...this.config.defaultOptions, ...options};
-
-    // If the content summary for this space has already been cached then it can be retrieved from there instead of making a network call to ediscovery
-    if (this.contentSummaryBySpaceIdCache.has(reportId) && this.contentSummaryBySpaceIdCache.get(reportId).has(spaceId)) {
-      return {body: this.contentSummaryBySpaceIdCache.get(reportId).get(spaceId), statusCode: 200};
-    }
-
-    this.logger.warn(`Cache miss for space ${spaceId} in getContentSummaryBySpaceId`);
-
-    return this.request({
-      method: 'GET',
-      url: `${reportUrl}/contents/summary/${spaceId}`,
-      timeout: options.timeoutMs
-    })
-      .then(tap((res) => this._writeSpacesToContentSummaryBySpaceIdCache(reportId, [res.body])));
+  clearCache() {
+    this.contentSummaryBySpaceIdCache.clear();
```

Comment:
This new API will need to be called to clear down the cache. Concern here is that reports with a large number of spaces could exhaust the app's RAM. But the largest reports we've come across will have < 10,000 spaces so it should not be an immediate concern. Still better to handle rather than let a bug/crash creep in.

Reply:
It's necessary for this change.

**tldr:** We use a space summary cache for performance. The previous code wiped this cache when the report id changed. In a federated report there can be multiple report ids meaning our cache was wiped when it should not have been.

With the upcoming federation support we're going to run a **local cluster** eDiscovery report. If it finds remote spaces it'll run, potentially several, **remote cluster** reports. But we hide these details behind a single report for the end user.

This change means that when you download a "single" report from ControlHub it can, under the covers, be made up of multiple report ids. Before there was only **one** report id. We took advantage of this and used it to manage our cache.

Namely when we detected the user was retrieving content for a new report id we cleared the existing cache. i.e. As the user was likely finished with those space summaries. But this no longer works with multiple report ids. Hence I'm making the call explicit.

We tend to `pre-populate` the cache for each report. That's important as it prevents the app from hammering the eDiscovery `getContentSummaryBySpaceId(..)` API. With federation we `pre-populate` all **local** and **remote** report spaces. But, as we clear when a new report id is detected, the code ended up only being populated with the details of the _last_ **remote report** id. To compensate for the missing cache details calls were being made to `getContentSummaryBySpaceId(..)`.